core/src/main/java/com/radixdlt/RadixNode.java

```diff
@@ -0,0 +1,212 @@
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+import com.radixdlt.api.CoreApiServer;
+import com.radixdlt.api.prometheus.PrometheusApi;
+import com.radixdlt.api.system.SystemApi;
+import com.radixdlt.consensus.bft.BFTValidatorId;
+import com.radixdlt.consensus.bft.Self;
+import com.radixdlt.consensus.safety.PersistentSafetyStateStore;
+import com.radixdlt.environment.Runners;
+import com.radixdlt.modules.ModuleRunner;
+import com.radixdlt.monitoring.MetricInstaller;
+import com.radixdlt.monitoring.Metrics;
+import com.radixdlt.networks.Network;
+import com.radixdlt.p2p.addressbook.AddressBookPersistence;
+import com.radixdlt.p2p.transport.PeerServerBootstrap;
+import com.radixdlt.statemanager.StateManager;
+import com.radixdlt.store.berkeley.BerkeleyDatabaseEnvironment;
+import com.radixdlt.transactions.RawLedgerTransaction;
+import com.radixdlt.utils.properties.RuntimeProperties;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public final class RadixNode {
+  private static final Logger log = LogManager.getLogger();
+
+  private final Injector injector;
+
+  private RadixNode(Injector injector) {
+    this.injector = injector;
+  }
+
+  public static RadixNode start(
+      RuntimeProperties properties, Network network, RawLedgerTransaction genesisTxn) {
+    log.info("Starting Radix node (genesis transaction: {})", genesisTxn.getPayloadHash());
+
+    final var injector = Guice.createInjector(new RadixNodeModule(properties, network, genesisTxn));
+
+    final var metrics = injector.getInstance(Metrics.class);
+    injector.getInstance(MetricInstaller.class).installAt(metrics);
+
+    final var moduleRunners =
+        injector.getInstance(Key.get(new TypeLiteral<Map<String, ModuleRunner>>() {}));
+
+    final var moduleStartOrder =
+        List.of(
+            Runners.P2P_NETWORK,
+            Runners.SYSTEM_INFO,
+            Runners.SYNC,
+            Runners.MEMPOOL,
+            Runners.CONSENSUS);
+
+    for (var module : moduleStartOrder) {
+      final var moduleRunner = moduleRunners.get(module);
+      moduleRunner.start();
+    }
+
+    final var peerServer = injector.getInstance(PeerServerBootstrap.class);
+    try {
+      peerServer.start();
+    } catch (InterruptedException e) {
+      log.error("Cannot start p2p server", e);
+      Thread.currentThread().interrupt();
```

Comment:
hmm, should the boot-up really continue after this?
Also, why is `InterruptedException` special in this regard? (I mean: if some other exception is thrown from `peerServer.start()`, it sweeps the callstack here, and definitely prevents the boot-up)

I suspect that you want only to "somehow" handle that awkward exception (because in practice it does not happen, or happens in a situation so dire [i.e. `ctrl+C` from user during boot-up], that we don't care how ungracefully the app ends). If that's the case, I recommend guava `Uninterruptibles` :p (but a good old "wrap as Runtime" works too).

Reply:
ha, I've just realized there is already an uninterruptible version of `sync` (`syncUninterruptibly`) in netty, so I went with that :)